pd/src/app.rs

```diff
@@ -107,17 +113,26 @@ impl App {
                 note::Note::try_from(note).expect("GenesisNote can be converted into regular Note"),
             );
         }
+
+        // Save `value_balance` so we can compute the binding verification key.
+        let value_balance = genesis_tx_builder.value_balance;
+
         let genesis_tx = genesis_tx_builder
             .set_chain_id(init_chain.chain_id)
             .finalize(&mut OsRng)
             .expect("can form genesis transaction");
+        let pending_transaction = genesis_tx
+            .verify_stateless(Some(value_balance))
+            .expect("genesis tx must be valid");
+        let verified_transaction = pending_transaction
+            .verify_stateful(self.note_commitment_tree.root2())
+            .expect("genesis tx must be valid");
 
         // Now add the transaction and its note fragments to the pending state changes.
-        genesis_block.add_transaction(genesis_tx);
+        genesis_block.add_transaction(verified_transaction);
```

Comment:
we verify the genesis transaction to produce `VerifiedTransaction` which now `PendingBlock::add_transaction` required